src/mono/mono/mini/aot-compiler.c

```diff
@@ -4546,6 +4550,39 @@ can_marshal_struct (MonoClass *klass)
 	return can_marshal;
 }
 
+/* Create a ref shared instantiation */
+static void
+create_ref_shared_inst (MonoAotCompile *acfg, MonoMethod *method, MonoGenericContext *ctx)
+{
+	MonoGenericContext shared_context;
+	MonoType **args;
+	MonoGenericInst *inst;
+	MonoGenericContainer *container;
+
+	memset (ctx, 0, sizeof (MonoGenericContext));
+
+	if (mono_class_is_gtd (method->klass)) {
+		shared_context = mono_class_get_generic_container (method->klass)->context;
+		inst = shared_context.class_inst;
+
+		args = g_new0 (MonoType*, inst->type_argc);
```

Comment:
Is the memory being deallocated correctly?

Reply:
No, it is not. I would suggest deallocating it after https://github.com/dotnet/runtime/blob/175dae98033a8a03627fbcdaa8575ad3a3fca512/src/mono/mono/metadata/metadata.c#L3399

Reply:
I can be freed, but the aot compiler is already leaking memory in a number of places, so its not that important.